compiler/semantic/proc.go

```diff
@@ -552,6 +552,20 @@ func semProc(ctx context.Context, scope *Scope, p ast.Proc, adaptor proc.DataAda
 			Type: typ,
 			As:   as,
 		}, nil
+	case *ast.Traverse:
+		e, err := semExpr(scope, p.Expr)
+		if err != nil {
+			return nil, err
+		}
+		seq, err := semSequential(ctx, scope, p.Seq, adaptor, head)
```

Comment:
At some point this should check that the resulting proc does not start with `From`.